### DISCUSSION TITLE
How to pass the current zoom level value to custom control component as a prop?

### DISCUSSION THREAD

**Discussion by dineshkumarrajaguru** [igordanchenko/yet-another-react-lightbox#215] [Q&A] [answered]
Hi Team,

I have created a custom control component with Zoom in and Zoom out button and by means of zoomRef I am invoking zoom in and zoom out methods, but I need to disable Zoom out button if the user has reached the zoom level 1.

I have tried the following options:

1. Using state to set the zoom level by listening to zoom event - But this is re-rendering the entire Lightbox and the first image is getting selected/shown.

2. By passing zoomRef?.current.zoom value to the CustomControl as a prop - But as we are using useRef the values are not getting updated.

Note: 
As I need to place the Zoom buttons at the bottom I couldn't use the render={{buttonZoom: ()=><CustomZoom />}}

@igordanchenko could you please guide me with this case?

I have created a simple component to illustrate the issue, please check [here](https://codesandbox.io/p/devbox/yet-another-react-lightbox-examples-forked-8hjtwq?file=%2Fsrc%2Fexamples%2FZoomPlugin.tsx&initialpath=%2Fplugins%2Fzoom&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpmvvw9z00092869lxmmrq93%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpmvvw9z00032869cftasbgg%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpmvvw9z00062869id2lyyp7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpmvvw9z00082869rm5p30v1%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpmvvw9z00032869cftasbgg%2522%253A%257B%2522id%2522%253A%2522clpmvvw9z00032869cftasbgg%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clpmvvw9z00082869rm5p30v1%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmvvw9z00072869cefqvpkm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252Fplugins%252Fzoomref%2522%257D%255D%252C%2522id%2522%253A%2522clpmvvw9z00082869rm5p30v1%2522%252C%2522activeTabId%2522%253A%2522clpmvvw9z00072869cefqvpkm%2522%257D%252C%2522clpmvvw9z00062869id2lyyp7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpmvvw9z000428692a68lut3%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clpmvvw9z00052869r9ovfcu9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clph9f4nu0007eghydk2f85if%2522%257D%255D%252C%2522id%2522%253A%2522clpmvvw9z00062869id2lyyp7%2522%252C%2522activeTabId%2522%253A%2522clpmvvw9z000428692a68lut3%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by igordanchenko** [comment, marked as answer]
Hi there!

Thank you for bringing up this use case. In retrospect, I should have probably exposed the `useZoom` hook in the Zoom plugin to make such a case more straightforward. However, here is how you can build the desired functionality with the existing features:

- track the `zoom` value with the `on.zoom` callback
- pass `zoom` and `zoomRef` as props to your custom control
- track the `index` value with the `on.view` callback if your slides reset to the initial slide (see [Tracking Slide Index](https://yet-another-react-lightbox.com/documentation#TrackingSlideIndex))

https://codesandbox.io/p/devbox/yet-another-react-lightbox-215-33hyhz?file=%2Fsrc%2FApp.tsx

I hope this helps!

**Comment by dineshkumarrajaguru** [comment]
Thank you for the solution, @igordanchenko . It greatly aided in addressing all the use cases for my implementation.
